src/daft-sql/src/modules/hashing.rs

```diff
@@ -0,0 +1,111 @@
+        inputs: &[FunctionArg],
+        planner: &crate::planner::SQLPlanner,
+    ) -> SQLPlannerResult<ExprRef> {
+        match inputs {
+            [input] => {
+                let input = planner.plan_function_arg(input)?;
+                Ok(hash(input, None))
+            }
+            [input, seed] => {
+                let input = planner.plan_function_arg(input)?;
+                match seed {
+                    FunctionArg::Named { name, arg, .. } if name.value == "seed" => {
+                        let seed = planner.try_unwrap_function_arg_expr(arg)?;
+                        Ok(hash(input, Some(seed)))
+                    }
+                    arg @ FunctionArg::Unnamed(_) => {
+                        let seed = planner.plan_function_arg(arg)?;
+                        Ok(hash(input, Some(seed)))
+                    }
+                    _ => unsupported_sql_err!("Invalid arguments for hash: '{inputs:?}'"),
+                }
+            }
+            _ => unsupported_sql_err!("Invalid arguments for hash: '{inputs:?}'"),
+        }
+    }
+}
+
+pub struct SQLMinhash;
+
+impl TryFrom<SQLFunctionArguments> for MinHashFunction {
+    type Error = PlannerError;
+
+    fn try_from(args: SQLFunctionArguments) -> Result<Self, Self::Error> {
+        let num_hashes = args
+            .get_named("num_hashes")
+            .ok_or_else(|| PlannerError::invalid_operation("num_hashes is required"))?
+            .as_literal()
+            .and_then(|lit| lit.as_i64())
+            .ok_or_else(|| PlannerError::invalid_operation("num_hashes must be an integer"))?
+            as usize;
+
+        let ngram_size = args
+            .get_named("ngram_size")
+            .ok_or_else(|| PlannerError::invalid_operation("ngram_size is required"))?
+            .as_literal()
+            .and_then(|lit| lit.as_i64())
+            .ok_or_else(|| PlannerError::invalid_operation("ngram_size must be an integer"))?
+            as usize;
+        let seed = args
+            .get_named("seed")
+            .map(|arg| {
+                arg.as_literal()
+                    .and_then(|lit| lit.as_i64())
+                    .ok_or_else(|| PlannerError::invalid_operation("num_hashes must be an integer"))
+            })
+            .transpose()?
+            .unwrap_or(1) as u32;
+        Ok(Self {
+            num_hashes,
+            ngram_size,
+            seed,
+        })
+    }
+}
+
+impl SQLFunction for SQLMinhash {
+    fn to_expr(
+        &self,
+        inputs: &[FunctionArg],
+        planner: &crate::planner::SQLPlanner,
+    ) -> SQLPlannerResult<ExprRef> {
+        match inputs {
+            [input, args @ ..] => {
+                let input = planner.plan_function_arg(input)?;
+                let args: MinHashFunction =
+                    planner.plan_function_args(args, &["num_hashes", "ngram_size", "seed"], 0)?;
+
+                Ok(minhash(input, args.num_hashes, args.ngram_size, args.seed))
+            }
```

Comment:
Could you explain this part to me, please? Why are we picking the first input and using that as the argument to `planner.plan_function_arg`?

Reply:
Oh. Are we specifically wanting 4 elements inside of that slice? Since the first one is `input`, the second, third, and fourth are `num_hashes`, `ngram_size`, and `seed`, respectively?

If so, should we match on specifically that case? I.e.:

```rs
match inputs {
  [input, num_hashes, ngram_size, seed] => ..,
  _ => unsupported_sql_err!(..),
}
```

Reply:
We do the pattern of `[input, args @ ..]` because some of these are optional, and they are not positional arguments, but instead named arguments. 

All of these are valid 
- `minhash(expr, num_hashes:=10, ngram_size:=10)`
- `minhash(expr, ngram_size:=10, num_hashes:=10)`
- `minhash(expr, seed:=1, ngram_size:=10, num_hashes:=10)`
- `minhash(expr, seed:=1, num_hashes:=10, ngram_size:=10)`

In this case, the only one that's positional is the input expr. So we match on `[input, args @ ..]`, then let `plan_function_args` handle the remainder.

Reply:
Oh okay, that makes sense to me. Thanks for explaining!